docs/design/gatewayapi-translator.md

```diff
@@ -0,0 +1,242 @@
+# Gateway API Translator Design
+
+## Assumptions
+- initially target core conformance features only, to be followed by extended conformance features
+
+## Inputs and Outputs
+
+The main inputs to the Gateway API translator are:
+- the GatewayClass to process
+- Gateways, HTTPRoutes, Services, Secrets
+
+The outputs of the Gateway API translator are:
+- IR
+- status updates for GatewayClass, Gateways, HTTPRoutes
+
+## Listener Compatibility
+Since Envoy Gateway handles all Gateways for a given GatewayClass, we need to determine the compatibility of _all_ Listeners across _all_ of those Gateways.
+
+The rules are:
+- for a given port number, every Listener using that port number must have a compatible protocol (either all HTTP, or all HTTPS/TLS).
+- for a given port number, every Listener using that port number must have a distinct hostname (at most one Listener per port can have no hostname).
+
+Listeners sharing a port that are not mutually compatible will be marked as "Conflicted: true" with an appropriate reason.
+
+### Listener Compatibility Examples
+
+#### Example 1: Gateways with compatible Listeners (same port & protocol, different hostnames)
+
+```yaml
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1beta1
+metadata:
+  name: gateway-1
+  namespace: envoy-gateway
+spec:
+  gatewayClassName: envoy-gateway
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+      allowedRoutes:
+        namespaces:
+          from: All
+      hostname: *.envoygateway.io
+---
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1beta1
+metadata:
+  name: gateway-2
+  namespace: envoy-gateway
+spec:
+  gatewayClassName: envoy-gateway
+  listeners:
+    - name: http
+      protocol: HTTP
+      port: 80
+      allowedRoutes:
+        namespaces:
+          from: All
+      hostname: whales.envoygateway.io
```

Comment:
this is a subset of `*.envoygateway.io`, does the spec mention this pattern for specific & regex hostnames ?

Reply:
Yep, wildcards are valid (https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Listener and https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.Hostname) but specific hostnames should take precedence over wildcards when matching.